File: apps/go-rest-api/services/redis_cloud_service.go
package services

import (
	"context"
	"fmt"
	"os"

	"github.com/redis/go-redis/v9"
)

var redisCloudClient *redis.Client

func InitRedisCloudClient() error {
	// Set Redis configuration
	address := os.Getenv("REDIS_ADDRESS")
	username := os.Getenv("REDIS_USERNAME")
	password := os.Getenv("REDIS_PASSWORD")

	// Initialize the Redis client
	client := redis.NewClient(&redis.Options{
		Addr:     address,
		Username: username,
		Password: password,
		DB:       0,
	})

	// Test the connection
	ctx := context.Background()
	_, err := client.Ping(ctx).Result()
	if err != nil {
		return fmt.Errorf("failed to connect to Redis: %v", err)
	}

	redisCloudClient = client
	return nil
}

func GetRedisCloudClient() *redis.Client {
	return redisCloudClient
}

File: apps/python-worker/services/redis_cloud_service.py
import redis
from config.settings import settings

def connect_to_redis_cloud():
    """Connect to Redis Cloud and return the client, or None on failure."""
    try:
        client = redis.Redis(
            host=settings.REDIS_ADDRESS,
            port=settings.REDIS_PORT,
            username=settings.REDIS_USERNAME,
            password=settings.REDIS_PASSWORD,
            decode_responses=True,
        )
        client.ping()
        print("Redis Cloud client initialized.")
        return client
    except redis.ConnectionError as e:
        print(f"Unable to connect to Redis Cloud: {e}")
    except Exception as e:
        print(f"Redis Cloud connection error: {e}")
    return None 